holoviews/ipython/parser.py

```diff
@@ -268,30 +313,39 @@ def parse(cls, line, ns={}):
             if (processed.strip() != line.strip()):
                 raise SyntaxError("Failed to parse remainder of string: %r" % line[e:])
 
+        grouped_paths = cls._group_paths_without_options(cls.opts_spec.parseString(line))
         parse = {}
-        for group in cls.opts_spec.parseString(line):
+        for pathspecs, group in grouped_paths:
             options = {}
 
             normalization = cls.process_normalization(group)
             if normalization is not None:
-                options['norm'] = Options(**normalization)
+                options['norm'] = normalization
 
             if 'plot_options' in group:
                 plotopts =  group['plot_options'][0]
                 opts = cls.todict(plotopts, 'brackets', ns=ns)
-                options['plot'] = Options(**{cls.aliases.get(k,k):v for k,v in opts.items()})
+                options['plot'] = {cls.aliases.get(k,k):v for k,v in opts.items()}
 
             if 'style_options' in group:
                 styleopts = group['style_options'][0]
                 opts = cls.todict(styleopts, 'parens', ns=ns)
-                options['style'] = Options(**{cls.aliases.get(k,k):v for k,v in opts.items()})
-
-            if group['pathspec'] in parse:
-                # Update in case same pathspec accidentally repeated by the user.
-                parse[group['pathspec']].update(options)
-            else:
-                parse[group['pathspec']] = options
-        return parse
+                options['style'] = {cls.aliases.get(k,k):v for k,v in opts.items()}
+
+            for pathspec in pathspecs:
+                if pathspec not in parse:
+                    parse[pathspec] = {}
+                cls._update_options(parse[pathspec], options)
```

Comment:
Perhaps something like ``parse[pathspec] = merged_options(parse.get(pathspec, {}), options)`` would be nicer?